src/modules/cpu/kernel/down_mixing.hpp

```diff
@@ -0,0 +1,148 @@
+    __m128 sums = _mm_add_ps(x, shuf);
+    shuf = _mm_movehl_ps(shuf, sums);        // high half -> low half
+    sums = _mm_add_ss(sums, shuf);
+    return _mm_cvtss_f32(sums);
+}
+
+Rpp32f rpp_hsum256_ps(__m256 x)
+{
+    __m128 p0 = _mm256_extractf128_ps(x, 1); // Contains x7, x6, x5, x4
+    __m128 p1 = _mm256_castps256_ps128(x);   // Contains x3, x2, x1, x0
+    __m128 sum = _mm_add_ps(p0, p1);         // Contains x3 + x7, x2 + x6, x1 + x5, x0 + x4
+    p0 = sum;                                // Contains -, -, x1 + x5, x0 + x4
+    p1 = _mm_movehl_ps(sum, sum);            // Contains -, -, x3 + x7, x2 + x6
+    sum = _mm_add_ps(p0, p1);                // Contains -, -, x1 + x3 + x5 + x7, x0 + x2 + x4 + x6
+    p0 = sum;                                // Contains -, -, -, x0 + x2 + x4 + x6
+    p1 = _mm_shuffle_ps(sum, sum, 0x1);      // Contains -, -, -, x1 + x3 + x5 + x7
+    sum = _mm_add_ss(p0, p1);                // Contains -, -, -, x0 + x1 + x2 + x3 + x4 + x5 + x6 + x7
+    return _mm_cvtss_f32(sum);
+}
+
+RppStatus down_mixing_host_tensor(Rpp32f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  Rpp32s *srcLengthTensor,
+                                  Rpp32s *channelsTensor,
+                                  bool normalizeWeights,
+                                  rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32s channels = channelsTensor[batchCount];
+        Rpp32s samples = srcLengthTensor[batchCount];
+        bool flag_avx = 0;
+
+        if(channels == 1)
+        {
+            // No need of downmixing, do a direct memcpy
+            memcpy(dstPtrTemp, srcPtrTemp, (size_t)(samples * sizeof(Rpp32f)));
+        }
+        else
+        {
+            Rpp32f *weights = static_cast<Rpp32f *>(malloc(channels * sizeof(Rpp32f)));
+            std::fill(weights, weights + channels, 1.f / channels);
+
+            if(normalizeWeights)
+            {
+                // Compute sum of the weights
+                Rpp32f sum = 0.0;
+                for(int i = 0; i < channels; i++)
+                    sum += weights[i];
+
+                // Normalize the weights
+                Rpp32f invSum = 1.0 / sum;
+                for(int i = 0; i < channels; i++)
+                    weights[i] *= invSum;
+            }
+
+            Rpp32s channelIncrement = 4;
+            Rpp32s alignedChannels = (channels / 4) * 4;
+
+            if(channels > 6)
+            {
+                flag_avx = 1;
```

Comment:
Why are we running the avx code only if channels > 6? Probably add some comment here to understand if anything is out of the ordinary.

Reply:
AVX can only work with 8 pixels in one go, so condition should be like if channels > 7 and not channels > 6.
If channels less than 7 then SSE or corresponding c code will be enabled. This was just for optimization purpose.

Reply:
@snehaa8 or @sampath1117 Could you please add this comment for clarity in the code?